src/NServiceBus.Core/DelayedDelivery/DelayedDelivery.cs

```diff
@@ -0,0 +1,92 @@
+﻿namespace NServiceBus.Features
+{
+    using System;
+    using NServiceBus.CircuitBreakers;
+    using NServiceBus.Config;
+    using NServiceBus.DelayedDelivery;
+    using NServiceBus.DelayedDelivery.TimeoutManager;
+    using NServiceBus.DeliveryConstraints;
+    using NServiceBus.Pipeline;
+    using NServiceBus.Timeout.Core;
+    using NServiceBus.TransportDispatch;
+    using NServiceBus.Transports;
+
+    /// <summary>
+    /// Allows to delay sent messages.
+    /// </summary>
+    public class DelayedDelivery : Feature
+    {
+        internal DelayedDelivery()
+        {
+            EnableByDefault();
+            //TM
+            Defaults(s => s.SetDefault("TimeToWaitBeforeTriggeringCriticalErrorForTimeoutPersisterReceiver", TimeSpan.FromSeconds(2)));
+
+            Prerequisite(c => c.DoesTransportSupportConstraint<DelayedDeliveryConstraint>()
```

Comment:
ok I understand the the prerequisites now. The Setup basically also handles each of the prerequisite checks differently to a certain degree. There are basically three scenarios which are coded into this feature:
* local timeout manager
* external timeout manager
* native delayed delivery by transport
It would probably make sense to make these scenarios a bit more explicit, but that's probably overkill.